src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -3354,6 +3470,66 @@ export function getUserGroupColor(
         : undefined;
 }
 
+export async function updateCustomIntervalFilter(
```

Comment:
i think this needs a high level comment on what exactly this does.

Reply:
it also could really use some unit testing : )

Reply:
yes, both are good ideas.